### PR TITLE
[py] Update WPEWebKit support code

### PR DESCRIPTION
**Pull request by lauromoura**
### Description

This PR changes the support code around WPEWebKit helper modules after the last update, a couple of years ago.

### Motivation and Context

In the recent Selenium versions, some changes in the way the driver classes are named and the move towards options instead of raw capabilities dictionaries, alongside other minor issues, made the existing WPEWebkit module to fail to create the driver correctly. For example, failing either to find the right service executable or failing the capability verification.

The WPEWebKit is a WebKit port focused on embedded devices. Developers creating WebApps for these devices can take advantage of the remote webdriver (used by these helper classes) to automate testing directly on the device.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
  - There's a change in a `wpewebkit.Service` parameter, but they're not yet covered by the API docs. Should I add them in a separate PR?
- [ ] I have updated the documentation accordingly.
  - Ditto
- [x] I have added tests to cover my changes.
  - The changes should be covered by the existing tests, so no new tests added.
- [x] All new and existing tests passed.
  - I didn't see any failure in the unit tests and most of the `//py:test-wpewebkit` passes, with some specific failures that will be dealt in a separate PR.

### CODE REVIEW

File: py/selenium/webdriver/common/driver_finder.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def get_path(service: Service, options: BaseOptions) -> str:
             msg = f"Unable to obtain driver for {options.capabilities['browserName']} using Selenium Manager."
             raise NoSuchDriverException(msg) from err
 
-        if path is None or not Path(path).is_file():
```
Comment (diemol) [inline py/selenium/webdriver/common/driver_finder.py]: Why is this needed?

File: py/selenium/webdriver/wpewebkit/webdriver.py [hunk 2]
```diff
@@ -48,18 +48,18 @@ def __init__(
...
-        if options:
-            capabilities = options.to_capabilities()
-            capabilities.update(desired_capabilities)
-            desired_capabilities = capabilities
-        else:
+        if not options:
             options = Options()
 
-        self.service = Service(executable_path, port=port, log_path=service_log_path)
+        if desired_capabilities:
```
Comment (titusfortner) [inline py/selenium/webdriver/wpewebkit/webdriver.py]: We shouldn't be supporting desired_capabilities in driver classes.

File: py/selenium/webdriver/common/driver_finder.py [hunk 3]
```diff
@@ -15,7 +15,7 @@
 # specific language governing permissions and limitations
 # under the License.
 import logging
-from pathlib import Path
+import shutil
```
Comment (titusfortner) [inline py/selenium/webdriver/common/driver_finder.py]: I think we should prefer pathlib to shutil. @isaulv / @AutomatedTester opinions?

File: py/selenium/webdriver/wpewebkit/webdriver.py [hunk 4]
```diff
@@ -48,18 +48,18 @@ def __init__(
...
+            for name, value in desired_capabilities.items():
+                options.set_capability(name, value)
+
+        self.service = Service(executable_path, port=port, log_output=service_log_path)
```
Comment (titusfortner) [inline py/selenium/webdriver/wpewebkit/webdriver.py]: A user should be able to pass in their own Service class, but it makes sense to require executable_path to be present for this class.

**Review by titusfortner — CHANGES_REQUESTED** [review]
To be honest, I think these classes should be in a separate library from Selenium. You should be able to subclass what is there and have it work properly. (If you can't, we should fix that). If it's in Selenium, it should match what the other drivers are doing.

File: py/selenium/webdriver/wpewebkit/webdriver.py (hunk 2 above)
Comment (lauromoura) [reply py/selenium/webdriver/wpewebkit/webdriver.py]: Right. I'll remove in favor of only the newer `options` scheme.

File: py/selenium/webdriver/wpewebkit/webdriver.py (hunk 4 above)
Comment (lauromoura) [reply py/selenium/webdriver/wpewebkit/webdriver.py]: Right. I'll update the `wpewebkit.WebDriver` API to match the Safari one, receiving a Service instead of the individual service details.

File: py/selenium/webdriver/common/driver_finder.py (hunk 3 above)
Comment (lauromoura) [reply py/selenium/webdriver/common/driver_finder.py]: I'll revert this change and make `wpewebkit.Service` use `shutil.which` internally for getting the right `WPEWebDriver` path as the default executable path.

File: py/selenium/webdriver/common/driver_finder.py (hunk 1 above)
Comment (lauromoura) [reply py/selenium/webdriver/common/driver_finder.py]: It should not be needed, I'm moving the path expansion into the driver-specific code.

File: py/selenium/webdriver/common/driver_finder.py (hunk 3 above)
Comment (sandeepsuryaprasad) [reply py/selenium/webdriver/common/driver_finder.py]: I think pathlib should be fine..
https://discuss.python.org/t/incrementally-move-high-level-path-operations-from-shutil-to-pathlib/19208

**Comment by lauromoura** [comment]
(Updated after missign a tox fix and rebase)

**Review by diemol — APPROVED** [review]

**Comment by lauromoura** [comment]
Pinging @titusfortner for review
